### PR TITLE
Update for mini_pupper_navigation

### PR DESCRIPTION
**Pull request by Hermanye996**
## Proposed changes
I have adjusted the cartographer slam.lua configuration with @0nhc, the base frame is /base_link now.
And the map for localization is mymap.pbstream which will be stored in /home directory if the user follows the new README.

These changes will make the navigation available on a real mini pupper.
### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code refactor (non-breaking change which refactors the code)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update only
- [ ] Other (please describe)

## Testing and Verification
To verify these changes, I performed the following tests:

1. Run the bringup launch file on Mini Pupper.
2. Run SLAM with a keyboard.
3. Save the map.
4. Run localization.
5. Run Navigation.
for more information, check the new README.

## Test Configuration

__Mini Pupper Version__  
Mini Pupper

__Raspberry Pi OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

__PC OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/mangdangroboticsclub/mini_pupper_ros/blob/ros2/CONTRIBUTING.md) guidelines.
- [X] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same change.

### CODE REVIEW

**Comment by Tiryoh** [comment]
Thanks @Hermanye996 
Could you run the following command to clarify the differences?

```
git checkout ros2_update_navigation
git revert 0c5bb5d
git revert 10fa184
git revert 03312f3
git push origin ros2_update_navigation
```

**Comment by Hermanye996** [comment]
> Thanks @Hermanye996 Could you run the following command to clarify the differences?
> 
> ```
> git checkout ros2_update_navigation
> git revert 0c5bb5d
> git revert 10fa184
> git revert 03312f3
> git push origin ros2_update_navigation
> ```

0c5bb5d & 10fa184 & 03312f3 have been deleted, this case was due to dependencies between multiple packages in update_branch_repos, thanks. @Tiryoh

File: mini_pupper_navigation/launch/localization.launch.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def generate_launch_description():
         DeclareLaunchArgument(
             'load_state_filename',
             default_value=os.path.join(
-                bringup_dir, 'maps', 'cartographer_map.pbstream'),
```
Comment (Tiryoh) [inline mini_pupper_navigation/launch/localization.launch.py]: Please either put the changes to the README regarding navigation in this pull request, or withdraw the changes here.

> And the map for localization is mymap.pbstream which will be stored in /home directory if the user follows the new README.

There is no `mymap.pbstream` file in the `mini_pupper_navigation/maps` folder.
This pull request does not contain the above document changes, so it is not consistent.

Comment (Hermanye996) [reply mini_pupper_navigation/launch/localization.launch.py]: I agree with you, mymap. pbstream file should not be put in the mini_pupper_navigation/maps folder. And I'll change it in README, users will save maps named cartographer_map.pbstream, which will replace official cartographer files in the maps folder, thanks.

File: README.md [hunk 2]
```diff
@@ -139,6 +139,61 @@ ros2 launch mini_pupper_bringup bringup.launch.py
...
+. ~/ros2_ws/install/setup.bash
+ros2 launch mini_pupper_bringup bringup.launch.py  # on real mini pupper
+ ```
+ - Replace the map files  
+ Remember to replace the cartographer_map.pbstream in the maps folder with your new cartographer_map.pbstream first.
+ ```bash
+ # Terminal 2 (on PC)
+ sudo cp -f ~/cartographer_map.pgm ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.pgm
+ sudo cp -f ~/cartographer_map.pbstream ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.pbstream
+ sudo cp -f ~/cartographer_map.yaml ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.yaml
```
Comment (Tiryoh) [inline README.md]:
```suggestion
cp -f ~/cartographer_map.pgm ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.pgm
cp -f ~/cartographer_map.pbstream ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.pbstream
cp -f ~/cartographer_map.yaml ~/ros2_ws/src/mini_pupper_ros/mini_pupper_navigation/maps/cartographer_map.yaml
```

Please avoid using sudo unless necessary. You should be able to operate everything under $HOME with basic user privileges.

File: README.md [hunk 3]
```diff
@@ -139,6 +139,61 @@ ros2 launch mini_pupper_bringup bringup.launch.py
...
+ ros2 launch mini_pupper_navigation slam.launch.py  # on PC
+ ```
+ - Keyboard control  
+Use the keyboard to remotely control the mini pupper to complete the mapping.
+```bash
+# Terminal 3 (on PC)
+ros2 run teleop_twist_keyboard teleop_twist_keyboard
+```
+- Save the map  
+If you want to save the map:
```
Comment (Tiryoh) [inline README.md]:
```suggestion
```

I think it is necessary to save the map for navigation, so how about save the map in general?

File: README.md (hunk 2 above)
Comment (Hermanye996) [reply README.md]: I agree with you that the abuse of sudo should be avoided.

**Review by Tiryoh — APPROVED** [review]

**Pull request merged**